### PR TITLE
Feature/3 create backend directory

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

* add the backend repository
* add some libraries
* add the DDD abstract classes

## Review points

This PR is very fat and the DDD classes are controversial in design.
So, please approve, and let's update it after implementation starts.

### CODE REVIEW

File: backend/main/test/Shared/DDDPrimitives/Identifier.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        "String": "nestedId",
+        "Number": 42
+    }
+}
+
+describe("string ID", () => {
+
+    const testId = TestIdentifierString.create("test");
+
+    test("equal: same string ID is false", () => {
```
Comment (taniguchiiqqq) [inline backend/main/test/Shared/DDDPrimitives/Identifier.test.ts:56]: I think this test name is supposed to be "different string ID is false".

File: backend/main/test/Shared/DDDPrimitives/Identifier.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+describe("number ID", () => {
+
+    const testId = TestIdentifierNumber.create(42);
+
+    test("equal: same number ID is true", () => {
+        const sameId = TestIdentifierNumber.create(42);
+        expect(testId.equal(sameId)).toBeTruthy();
+    })
+
+    test("equal: same number ID is true", () => {
```
Comment (taniguchiiqqq) [inline backend/main/test/Shared/DDDPrimitives/Identifier.test.ts:76]: I think this test name is supposed to be "different numberID is false".

File: backend/main/test/Shared/DDDPrimitives/Identifier.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        expect(testId.id.Object.Number).toBe(testParam.Object.Number);
+        expect(testId.id).toEqual(testParam);
+    })
+
+    test("equal: same object ID is true", () => {
+        const sameId = TestIdentifierObject.create(testParam);
+        expect(testId.equal(sameId)).toBeTruthy();
+    })
+
+    test("equal: same object ID is true", () => {
```
Comment (taniguchiiqqq) [inline backend/main/test/Shared/DDDPrimitives/Identifier.test.ts:100]: I think this test name is supposed to be "different object ID is false".

File: backend/main/test/Shared/testData/identifierTestData.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export class IdentificationTestClass extends Identifier<number> {
+    static create(id: number): IdentificationTestClass {
+        return new IdentificationTestClass(id)
+    }
+}
+
+/**
+ * for test use, id 42
+ */
+export const identificationTestData = IdentificationTestClass.create(42);
```
Comment (taniguchiiqqq) [inline backend/main/test/Shared/testData/identifierTestData.ts]: From just curiosity, where will you use this identificationTestData ?

**Review by taniguchiiqqq — APPROVED** [review]
LGTM with Nits.

**Comment by hitohata** [comment]
@taniguchiiqqq 
Thank you for your review. I fixed the test comment you mentioned.
I merged this branch.

**Pull request merged**